### PR TITLE
WIP: Fixing RI issues on windows

### LINKED ISSUE #2545: RI handling fails on Windows
It should have 

```
assert_raises(ValueError, clone, 'Zoidberg', path='ssh://mars:Zoidberg')
```

But did:

```
======================================================================
380ERROR: datalad.distribution.tests.test_clone.test_invalid_args
381----------------------------------------------------------------------
382Traceback (most recent call last):
383  File "C:\Miniconda35\envs\test-environment\lib\site-packages\nose\case.py", line 198, in runTest
384    self.test(*self.arg)
385  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
386    return t(*(arg + (filename,)), **kw)
387  File "C:\Miniconda35\envs\test-environment\lib\contextlib.py", line 77, in __exit__
388    self.gen.throw(type, value, traceback)
389  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\utils.py", line 1307, in make_tempfile
390    yield filename
391  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
392    return t(*(arg + (filename,)), **kw)
393  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
394    return t(*(arg + (filename,)), **kw)
395  File "C:\Miniconda35\envs\test-environment\lib\contextlib.py", line 77, in __exit__
396    self.gen.throw(type, value, traceback)
397  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\utils.py", line 1307, in make_tempfile
398    yield filename
399  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
400    return t(*(arg + (filename,)), **kw)
401  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
402    return t(*(arg + (filename,)), **kw)
403  File "C:\Miniconda35\envs\test-environment\lib\contextlib.py", line 77, in __exit__
404    self.gen.throw(type, value, traceback)
405  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\utils.py", line 1307, in make_tempfile
406    yield filename
407  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
408    return t(*(arg + (filename,)), **kw)
409  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\distribution\tests\test_clone.py", line 71, in test_invalid_args
410    assert_raises(ValueError, clone, 'Zoidberg', path='ssh://mars:Zoidberg')
411  File "C:\Miniconda35\envs\test-environment\lib\unittest\case.py", line 727, in assertRaises
412    return context.handle('assertRaises', args, kwargs)
413  File "C:\Miniconda35\envs\test-environment\lib\unittest\case.py", line 176, in handle
414    callable_obj(*args, **kwargs)
415  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\interface\utils.py", line 470, in eval_func
416    return return_func(generator_func)(*args, **kwargs)
417  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\interface\utils.py", line 458, in return_func
418    results = list(results)
419  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\interface\utils.py", line 446, in generator_func
420    msg="Command did not complete successfully")
421datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully [{'status': 'error', 'type': 'dataset', 'source_url': 'Zoidberg', 'message': ('Failed to clone data from any candidate source URL: %s', ['Zoidberg']), 'action': 'install', 'path': 'C:\\projects\\datalad-5sgef\\__testhome__\\ssh:\\mars:Zoidberg'}]
```

### PR DESCRIPTION
**Pull request by bpoldrack**
This pull request fixes #2545 

This pull request proposes

### Changes
- [ ] This change is complete

Please have a look @datalad/developers

### CODE REVIEW

**Comment by mih** [comment]
Is there anything in here that still needs to get into master?

**Comment by mih** [comment]
I have rebased this PR on the current master to see what is still new.

File: datalad/support/sshconnector.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ def open(self):
         # start control master:
         lgr.debug("Opening %s by calling %s" % (self, cmd))
         proc = Popen(cmd)
-        stdout, stderr = proc.communicate(input="\n")  # why the f.. this is necessary?
+        stdout, stderr = proc.communicate(input="%s" % linesep)  # why the f.. this is necessary?
```
Comment (mih) [inline datalad/support/sshconnector.py]: Just: `input=linesep`?

**Comment by yarikoptic** [comment]
#2545 is already marked fixed, so what should be the sentence for this PR?

**Comment by yarikoptic** [comment]
I  will close it for now, @bpoldrack feel welcome to reopen if anything to be salvaged here

**Pull request closed without merging**